decidim-proposals/app/commands/decidim/proposals/endorse_proposal.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    # A command with all the business logic when a user endorses a proposal.
+    class EndorseProposal < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # proposal     - A Decidim::Proposals::Proposal object.
+      # current_user - The current user.
+      # current_group_id- (optional) The current_grup that is endorsing the Proposal.
+      def initialize(proposal, current_user, current_group_id = nil)
+        @proposal = proposal
+        @current_user = current_user
+        @current_group_id = current_group_id
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the proposal vote.
+      # - :invalid if the form wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        build_proposal_endorsement
+        return broadcast(:invalid) unless @endorsement.valid?
+        @endorsement.save!
+        broadcast(:ok, endorsement)
+      end
+
+      attr_reader :endorsement
+
+      private
+
+      def build_proposal_endorsement
+        @endorsement = @proposal.endorsements.build(author: @current_user)
```

Comment:
@xabier should the followers of the author get a notification about this?

Reply:
https://github.com/decidim/decidim/issues/2287#issuecomment-356866768

Reply:
@oriolgual can you accept this PR without notifications and then I'll add notification features in a new PR?
I already took note of all notification related comments at #2287 for next sprint